File: Reqnroll/Chapter15/15.7-TheStubDependencyPattern/After/WIMP.Specs/Drivers/TimeServiceDriver.cs
using WIMP.Specs.Support;

namespace WIMP.Specs.Drivers;

public class TimeServiceDriver(StubTimeService stubTimeService)
{
    public void SetCurrentTime(TimeOnly time)
    {
        var dateTime = stubTimeService.GetCurrentTime().WithTime(time);
        stubTimeService.SetCurrentTime(dateTime);
    }

    public DateTimeOffset GetTodayTime(TimeOnly time)
    {
        return stubTimeService.GetCurrentTime().WithTime(time);
    }
}

File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/After/src/test/java/com/wimp/app/specs/drivers/TimeServiceDriver.java
package com.wimp.app.specs.drivers;

import com.wimp.app.specs.support.StubTimeService;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.time.LocalTime;

@Component
public class TimeServiceDriver {
    private final StubTimeService stubTimeService;

    public TimeServiceDriver(StubTimeService stubTimeService) {
        this.stubTimeService = stubTimeService;
    }

    public void setCurrentTime(LocalTime time) {
        LocalDateTime dateTime = getTodayTime(time);
        stubTimeService.setCurrentTime(dateTime);
    }

    public LocalDateTime getTodayTime(LocalTime time) {
        return stubTimeService.getCurrentTime().with(time);
    }
}
